pkg/server/plugin/datastore/sql/sql.go

```diff
@@ -1368,25 +1368,34 @@ func listRegistrationEntriesOnce(ctx context.Context, dbType string, db *sql.DB,
 	}
 	var lastEID uint64
 
+	pushEntry := func(entry *common.RegistrationEntry) {
+		// Due to previous bugs, there can be cruft rows related to a deleted
+		// registration entries that are fetched with the list query. To avoid
+		// hydrating partial entries, append only entries that have data from
+		// the registered_entries table (i.e. those with an entry id).
+		if entry != nil && entry.EntryId != "" {
+			entries = append(entries, entry)
+		}
+	}
+
+	var entry *common.RegistrationEntry
```

Comment:
could you have `pushEntry` (or maybe it would get a different name) close over `entry` and `lastEID` so that it can handle all of this logic?

It would be nice to avoid duplicating 
```
if entry != nil && entry.EntryId != "" {
```

Reply:
I went to do this and then backed it out. It felt like it reduced readability, since it required moving the row struct out of the loop (so it can be captured), which meant adding code to zero the row struct on every loop before filling from the query row. It also felt weird to run code that updated the lastEID outside of the loop where it wasn't really applicable. I'm welcome to revisit if you feel strongly, but otherwise I think I'd rather leave this as-is.

Reply:
:+1: